app/src/main/java/org/wikipedia/readinglist/recommended/RecommendedReadingListViewModel.kt

```diff
@@ -19,7 +19,7 @@ import org.wikipedia.util.Resource
 
 class RecommendedReadingListViewModel(savedStateHandle: SavedStateHandle) : ViewModel() {
 
-    val fromSettings = savedStateHandle.get<Boolean>(Constants.ARG_BOOLEAN) ?: false
+    val fromSettings = savedStateHandle.get<Boolean>(RecommendedReadingListOnboardingActivity.EXTRA_FROM_SETTINGS) == true
```

Comment:
Do you want to hide the bottom dice icon and the arrow icon if it comes from the settings? and also keep the layout to the topbar. You may check the `SourceSelectionScreen` for a similar logic for handling.